src/cryptonote_basic/cryptonote_format_utils.cpp

```diff
@@ -919,6 +921,51 @@ namespace cryptonote
     return s;
   }
   //---------------------------------------------------------------
+  char const *print_tx_verification_context(tx_verification_context const &tvc, transaction const *tx)
+  {
+    static char buf[1024];
```

Comment:
This buffer is not big enough if all of the errors are triggered. I know it will never happen, but just so you know.

The typical way to do this is C++ is `std::stringstream ss`:

    ss << "New message";
    ss << "new message blah blah blah";
    ss.str(); // get the resulting string.